### ISSUE TITLE
wind arrow direction depends on map projection

### ISSUE THREAD

**Issue by jgrooss** [Open-MSS/MSS#1899]
When displying plots ECMWF wind data arrows the wind file direction is not correct in all times. The following plots show the same polar stereogaphic plots but for opposite mirrored projections.
For comparison I attach also a gaphically mirrired file.
v1 shows a projection an a flight path from EDMO (upper edge) to Anchorage (lower edge) 
map frame settings are  "llcrnrlon": 150.0, "llcrnrlat": 50.0, "urcrnrlon": -45.0, "urcrnrlat": 30.0
This seems to be not correct

v2 shows the same data with EDMO at the lower edge
map frame settings are  "llcrnrlon": 60.0, "llcrnrlat": 32.0, "urcrnrlon": -127.0,,  "urcrnrlat": 48.0
This seems to be correct.

For visual comparison, I also add a mirrored image of V1
**V1:**
![topview_Geopotential_wind_150hPa_v1](https://github.com/Open-MSS/MSS/assets/32653190/88b7f52e-4022-4a6d-8d42-15e839a4ac72)
**V2:**
![topview_Geopotential_wind_150hPa_v2](https://github.com/Open-MSS/MSS/assets/32653190/8e07905e-a453-4027-a3f6-e0dffa00aa73)
**V1-mirror:**
![topview_Geopotential_wind_150hPa_v1_mirror](https://github.com/Open-MSS/MSS/assets/32653190/719fa9ab-398e-42ce-85fc-c7afd95e4a3b)

**Comment by jgrooss** [comment]
@joernu76 Do you have any idea what causes this behaviour?

**Comment by jgrooss** [comment]
The cylindrical projection would be likely correct, This contributes better to V2

![topview_Geopotential_wind_150hPa_cyl](https://github.com/Open-MSS/MSS/assets/32653190/254fe339-daa1-4fc0-a22c-20253795221e)

**Comment by jgrooss** [comment]
NB this is very important as we tried to get a hint whether a direct flight would be possible due to head winds or not...
If the wind arrow  depiction does not work for stereographic projections, it should rather left out in this projection.

**Comment by ReimarBauer** [comment]
We don't use cartopy, but that sounds similiar
https://github.com/SciTools/cartopy/issues/1179

look for the section, comparison basemap to cartopy
```
m = basemap.Basemap(projection="npstere", boundinglat = 50,  lon_0 = 0)
u_rot, v_rot, x, y = m.rotate_vector(u, v, lon ,lat, returnxy=True)
m.quiver(x, y, u_rot, v_rot, angles = "xy")
```

the rotate_vector is mentioned as solution. This needs to become fixed in mpl_hsec_styles

**Comment by joernu76** [comment]
Moin, Ill try to have a look. The projection in V2 is mirrored, which happens with "wrong" lon/lat coordinates even though. This is
The wind arrows look fishy, though. Wind should go from the Americas to Europe, I guess.

**Comment by joernu76** [comment]
This projection works for me:
```
        "16 Northern Hemisphere (stereo)": {                                                                                                                                 
            "CRS": "EPSG:77789225",
            "map": {
                "llcrnrlon": 180.0,
                "llcrnrlat": 0.0,
                "urcrnrlon": 0.0,
                "urcrnrlat": 0.0
            }
        }   
```

**Comment by ReimarBauer** [comment]
@jgrooss is your question answered?

**Comment by jgrooss** [comment]
No,  the wind arrows should work for every chosen projection. It seems to me however, that this is an error in the included plotting routines

**Comment by joernu76** [comment]
Correct, but if the projection is wrongly parameterized, i.e. if, e.g., the landmasses are reversed in east-west order, nothing is assured to work anymore.

In particular, you are using a projection that is centered on 10°E and 40°S. The coordinates of the corners are flipped in such a fashion that it looks fine, but the winds are wrong.

Incidentally, the "02 PHILEAS (stereo)" projection has the same issue. Using an appropriate center of projection, the winds are fine as well:
```
        "02 PHILEAS (stereo)": {
            "CRS": "EPSG:77740190",
            "map": {
                "llcrnrlon": 170.0,
                "llcrnrlat": 40.0,
                "urcrnrlon": 260.0,
                "urcrnrlat": 65.0
            }
        },
```
The specification of corners in lon/lat is dangerous with basemap stereographic projections...

**Comment by joernu76** [comment]
I uploaded a fixed msui_settings.json file to the PHILEAS cloud. I also added better projections for Norther Canada and a polar Northern Hemisphere plot.

That something is seriously wrong with the current projection is apparent when one zooms out. The area around Anchorage is severely distorted due to the large distance to the center of projection.
